src/qt/optionsmodel.cpp

```diff
@@ -344,15 +356,19 @@ bool OptionsModel::setData(const QModelIndex & index, const QVariant & value, in
         case HideTrayIcon:
             fHideTrayIcon = value.toBool();
             settings.setValue("fHideTrayIcon", fHideTrayIcon);
-    		Q_EMIT hideTrayIconChanged(fHideTrayIcon);
+            Q_EMIT hideTrayIconChanged(fHideTrayIcon);
             break;
         case MinimizeToTray:
             fMinimizeToTray = value.toBool();
             settings.setValue("fMinimizeToTray", fMinimizeToTray);
             break;
+        case MapPortNatpmp: // core option - can be changed on-the-fly
+            settings.setValue("fUseNatpmp", value.toBool());
+            m_node.mapPort(value.toBool(), settings.value("fUseUPnP").toBool());
```

Comment:
This logic breaks the "command line overrides GUI options" stuff above (see `addOverriddenOption`)...

Not really sure what the best solution here is.

Reply:
The separate issue #18184 has been submitted.